ocaml/xapi/xapi_host.ml

```diff
@@ -488,8 +488,17 @@ let shutdown_and_reboot_common ~__context ~host label description operation cmd
   if Db.Host.get_enabled ~__context ~self:host
   then raise (Api_errors.Server_error (Api_errors.host_not_disabled, []));
 
+  if (Pool_role.is_master ()) then
+       (* We are the master and we are about to shutdown HA and redo log:
+          prevent slaves from sending (DB) requests.
+          If we are the slave we cannot shutdown the request thread yet
+          because we might need it when unplugging the PBDs
+        *)
+       Remote_requests.stop_request_thread();
+
   Xapi_ha.before_clean_shutdown_or_reboot ~__context ~host;
   Xapi_pbd.unplug_all_pbds ~__context;
+
   Remote_requests.stop_request_thread();
```

Comment:
Is it OK to call this twice?

Reply:
Should be: it sets a a variable and signals a condition, I pushed a fixup to do the `stop_request_thread` only once though, just in case we  change that in the future.